inc/savedsearch.class.php

```diff
@@ -0,0 +1,1370 @@
+      }
+      return parent::canViewItem();
+   }
+
+
+   function isNewItem() {
+      /// For tabs management : force isNewItem
+      return false;
+   }
+
+
+   function defineTabs($options=array()) {
+      $ong = [];
+      $this->addDefaultFormTab($ong);
+      return $ong;
+   }
+
+
+   function getSearchOptionsNew() {
+      global $CFG_GLPI;
+
+      $tab = [];
+
+      $tab[] = [
+         'id'                 => 'common',
+         'name'               => __('Characteristics')
+      ];
+
+      $tab[] = [
+         'id'                 => '1',
+         'table'              => $this->getTable(),
+         'field'              => 'name',
+         'name'               => __('Name'),
+         'datatype'           => 'itemlink',
+         'massiveaction'      => false // implicit key==1
+      ];
+
+      $tab[] = [
+         'id'                 => '2',
+         'table'              => $this->getTable(),
+         'field'              => 'id',
+         'name'               => __('ID'),
+         'massiveaction'      => false, // implicit field is id
+         'datatype'           => 'number'
+      ];
+
+      $tab[] = [
+         'id'                 => 3,
+         'table'              => User::getTable(),
+         'field'              => 'name',
+         'name'               => __('User'),
+         'datatype'           => 'dropdown'
+      ];
+
+      $tab[] = [
+         'id'                 => '8',
+         'table'              => $this->getTable(),
+         'field'              => 'itemtype',
+         'name'               => __('Item type'),
+         'massiveaction'      => false,
+         'datatype'           => 'itemtypename',
+         'types'              => self::getUsedItemtypes()
+      ];
+
+      $tab[] = [
+         'id'                 => 9,
+         'table'              => $this->getTable(),
+         'field'              => 'last_execution_time',
+         'name'               => _('Last duration (ms)'),
+         'massiveaction'      => false,
+         'datatype'           => 'number'
+      ];
+
+      $tab[] = [
+         'id'                 => 10,
+         'table'              => $this->getTable(),
+         'field'              => 'do_count',
+         'name'               => _('Count'),
+         'massiveaction'      => true,
+         'datatype'           => 'specific',
+         'searchtype'         => 'equals'
+      ];
+
+      $tab[] = [
+         'id'                 => 11,
+         'table'              => SavedSearch_User::getTable(),
+         'field'              => 'users_id',
+         'name'               => __('Default'),
+         'massiveaction'      => false,
+         'joinparams'         => [
+            'jointype'           => 'child'
+         ],
+         'datatype'           => 'specific',
+         'searchtype'         => [
+            0 => 'equals',
+            1 => 'notequals'
+         ]
+      ];
+
```

Comment:
it could be usefull to have an additional option for execution time (datatype timestamp)